### PR TITLE
Maybe overrides more parent methods

### LINKED ISSUE #2436: Maybe helpfile example code doesn't work as described
Thanks to simdax:

```
(
   p.pop.clear;
   p = LazyEnvir.new;
   p.proxyClass = Maybe;
   p.linkDoc; // here: connect to current doc only.
);


// envirs
~a = (note: [1, 2]);
~b = (dur: 1);
~c = ~a.putAll(~b) // provisionally put everything into the placholder
~c.value;
~a = (note: [1, 2, 4]);
~c.value;
~d = ~a.at(\note);
~d.value;
~a = (note: [7.5]);
~d.value; // [7.5]  

====> give to me [1,2,4] and *not* [7.5]

```

### PR DESCRIPTION
**Pull request by telephon**
This fixes #2436

### CODE REVIEW

File: HelpSource/Classes/Maybe.schelp [hunk 1]
```diff
@@ -8,6 +8,23 @@ A Maybe object can contain either nil or some other object, and allows to constr
...
+You can extend Maybe as you need for your purpose, by the following schema:
+
+code::
+// replaye "myMethod" with your method name
++ Maybe {
+	myMethod { |...args|
+		^this.composeNAryOp(\myMethod, args)
+	}
+}
+::
```
Comment (snappizz) [inline HelpSource/Classes/Maybe.schelp]: IMO extensions to built-in classes are a bad habit. I'm not necessarily saying that you should remove this example, but maybe warn of the possible dangers.

**Comment by telephon** [comment]
I think this is fine. Opinions?

**Comment by telephon** [comment]
ping

**Comment by snappizz** [comment]
honk honk, anyone else want to look at this?

**Pull request merged**